frontend/packages/data-portal/app/components/BrowseData/DatasetTable.tsx

```diff
@@ -43,13 +45,24 @@ export function DatasetTable() {
   const { datasets } = useDatasets()
 
   const [searchParams, setSearchParams] = useSearchParams()
+  const { setBrowseAllHistory } = useFilterHistory()
   const datasetSort = (searchParams.get('sort') ?? undefined) as
     | CellHeaderDirection
     | undefined
 
   const { isLoadingDebounced } = useIsLoading()
 
-  const location = useLocation()
+  useEffect(
+    () =>
+      setBrowseAllHistory(
+        new Map(
+          Array.from(searchParams).filter(([k]) =>
+            DATASET_FILTERS.map((v) => v as string).includes(k),
```

Comment:
@codemonkey800 is there a better way of asserting this is an array of strings to typescript? I had issues doing `DATASET_FILTERS as string[]` since it would complain

Reply:
it complains about this as well unfortunately since `BrowseAllHistoryKey` is a set of constant strings